### ISSUE TITLE
Allow Customers to use dates in API

### ISSUE THREAD

**Issue by cklosowski** [easydigitaldownloads/easy-digital-downloads#3157]
As requested in https://easydigitaldownloads.com/support/topic/need-api-features-adding-to-edd/

Would be nice to have the `customers` data in the API to use our date ranges to search by the `date_created`.

**Comment by pippinsplugins** [comment]
:+1:

**Comment by chriscct7** [comment]
Dependent on #3337

**Comment by chriscct7** [comment]
(for backwards compatibility reasons)

**Comment by pippinsplugins** [comment]
Punting this to 2.5

**Comment by cklosowski** [comment]
:+1:

**Comment by cklosowski** [comment]
@pippinsplugins since we don't have the week ranges currently in the API, is this good to go? We can look at mirroring the Reporting date ranges in #2462 possibly?

**Comment by pippinsplugins** [comment]
Let's go ahead and mirror them.

**Comment by cklosowski** [comment]
@pippinsplugins adding `this_week` and `last_week` is a far larger project than adding dates to the customers endpoint. I propose we merge customers endpoint with the existing date ranges, and then when we come in to revamp the dates part of the API we rely on the stats class dates parsing as it's far more flexible. No sense in mirroring them in this branch just to have them redone in another.

**Comment by pippinsplugins** [comment]
That's fine by me.

**Comment by pippinsplugins** [comment]
Just added a fix that allows customers to be shown even if no date is specified. IT was returning empty results if no date was specified.
